src/rpc/methods/eth/types.rs

```diff
@@ -309,6 +311,44 @@ impl TryFrom<EthCallMessage> for Message {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema, PartialEq, Eq, Hash)]
+pub struct EthHash(#[schemars(with = "String")] pub H256);
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema, PartialEq, Eq, Hash, Clone)]
+pub struct FilterID(EthHash);
+
+lotus_json_with_self!(FilterID);
+
+impl FilterID {
+    pub fn new() -> Result<Self, uuid::Error> {
+        let raw_id = Uuid::new_v4();
+        let mut id = [0u8; 32];
+        id[..16].copy_from_slice(raw_id.as_bytes());
+        Ok(FilterID(EthHash(H256::from_slice(&id))))
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+pub struct EthHashList(pub Vec<EthHash>);
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+pub struct EthTopicSpec(pub Vec<EthHashList>);
+
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct EthFilterSpec {
```

Comment:
It would be great to have documentation for this type and some links to the given specification.

Reply:
added